### ISSUE TITLE
VPN Bridge - Routing problem

### ISSUE THREAD

**Issue by mkaand** [hwdsl2/setup-ipsec-vpn#400]
Hello,

Thank you very much for this great script. I successfully made connection between my networks. But I have routing problem. I hope you can help me. Here is the setup:

My VPN Server on my Scaleway VPS. I have public IP and here is the ifconfig results:

`root@scw-5697a4:~# ifconfig
ens2      Link encap:Ethernet  HWaddr de:1a:0c:25:f0:0d
          inet addr:10.14.136.25  Bcast:10.14.136.25  Mask:255.255.255.254
          inet6 addr: fe80::dc1a:cff:fe25:f00d/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:11309 errors:0 dropped:0 overruns:0 frame:0
          TX packets:9898 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:62779095 (62.7 MB)  TX bytes:1189404 (1.1 MB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.0.0.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:5 errors:0 dropped:0 overruns:0 frame:0
          TX packets:5 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:325 (325.0 B)  TX bytes:325 (325.0 B)

ppp0      Link encap:Point-to-Point Protocol
          inet addr:192.168.42.1  P-t-P:192.168.42.10  Mask:255.255.255.255
          UP POINTOPOINT RUNNING NOARP MULTICAST  MTU:1280  Metric:1
          RX packets:26 errors:0 dropped:0 overruns:0 frame:0
          TX packets:18 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:3
          RX bytes:7376 (7.3 KB)  TX bytes:5060 (5.0 KB)
`

and this is route table:

`Kernel IP routing table
Destination     Gateway         Genmask         Flags Metric Ref    Use Iface

default         10.14.136.24    0.0.0.0         UG    0      0        0 ens2
10.14.136.24    *               255.255.255.254 U     0      0        0 ens2
192.168.42.10   *               255.255.255.255 UH    0      0        0 ppp0
`

CLIENT SIDE I use Raspberry Pi Zero
Client Side Network: 192.168.2.0 and RaspberryPi has static IP 192.168.2.3
Client VPN IP: 192.168.42.10

I can ping each other but from VPN SERVER I cannot ping Client Side route 192.168.2.1 
I should able to access from VPN Server to VPN Client's network. It seems routing problem but I am not expert.

**Comment by mkaand** [comment]
If I add to following route to VPN SERVER:
route add -net 192.168.2.0/24 gw 192.168.42.10

I can ping 192.168.2.3 (VPN Client local IP) but I cannot ping Client router (192.168.2.1)
If I traceroute 192.168.2.1 I t goes and stop at 192.168.42.10. I guess my problem on VPN Client (Raspberry side) I hope someone can help me. What I want:

I have VPN Server on VCS Scaleway. I have public IP let say X.X.X.X
My VPN client is raspberery and behind several routers.

Remote (Client Side) router: 192.168.2.1
IP Camera: 192.168.2.2
Raspberry: 192.168.2.3

I made VPN Connection to Raspberry. But I cannot route traffic to router. I should able to access my webcam via some special ports like rtsp etc from X.X.X.X

**Comment by hwdsl2** [comment]
@mkaand Hello! For IPsec VPNs it is generally difficult to access from the VPN Server to the VPN Client's network (the other way around is usually fine), unless the two endpoints are connected as subnet-to-subnet VPNs [1]. I suggest that you instead seek help from the Libreswan or strongSwan users mailing lists [2] [3].

[1] https://libreswan.org/wiki/Configuration_examples
[2] https://lists.libreswan.org/mailman/listinfo/swan
[3] https://lists.strongswan.org/mailman/listinfo/users

**Comment by mkaand** [comment]
Hello,

Is there anyway to create a L2TP Server without PSK (IPSEC) I mean I have one Edimax BR6258n router. It supports L2TP. I want to use it. But I have SoftEther VPN Server at home. It uses PSK but BR6258n client doesn't have option to type KEY (Secret). What I want to:

I want to create simple L2TP VPN server on ubuntu. And I want to connect via BR6258n mini router without PSK. How can I do that?

**Comment by hwdsl2** [comment]
@mkaand Hello! Using L2TP without IPsec means that the traffic is not encrypted at all, which defeats the purpose of using a VPN. For this reason, the IPTables rules in the VPN scripts disallow L2TP without IPsec connections to UDP port 1701.

**Comment by mkaand** [comment]
Yes you right but most of routers DDWRT and other in L2TP section there is no PSK field. My iPhone can connect to my SoftEther VPN Server. iPhone has secret key option in L2TP VPN options. On the other hands I see SoftEther VPN Server allow to use L2TP without IPSec but still I cannot make connection from outside of my network. It seems some firewalls blocks as you said. Because from outside I ca connect to my Home VPN Server with IPSec. SEcurity is not important for me. I have hobby garden and there is public wifi there. I have another router at my garden. This router uses public wifi and my webcam goes internet via this router. But I need direct access to my webcam for grabbing streams. I used use OpenVPN for that and it was perfect network. But I guess public wifi changed something they added extra firewalls or they reduced the speed I don't know but OpenVPN gives TLS handshake errors. That's why I am looking for alternative. I cannot make connection from my garden via OpenVPN. I have VPC Server too I tried everything but no luck. But I tested my iPhone on at my garden. It connects perfectly to my home SoftEther VPN Server via L2TP/IPSec. By the way there is Raspberry Pi Zero Wifi in my Garden too. RP checks internet connection in every minute if there is no internet access or riched the daily limit, my shell script runs and automatically switch captive portal login. So what I have:

Main router at home WRT1900AC OpenVPN Server + SoftEther VPN Server DDWRT
Garden Router: TP-LINK DDWRT
Garden RaspberryPi
Garden WebCamera
Garden Public Wifi (I do not control this network)
And if you offer me better solution I have some servers on scaleway (VPC)

What I want: 

I should able to direct access some special ports of Webcamera from internet.

My past network:

Camera -> NAT ---- Garden Router --- OpenVPN Client <--INTERNET--->  OpenVPN Server (Main Router) -- NAT ---> Smart TV

I mean all traffic from my garden goes thru my home. But not anymore. Because of TLS handshake errors.
